OpenSim/Simulation/Model/Model.cpp

```diff
@@ -659,34 +657,42 @@ void Model::extendConnectToModel(Model &model)
 
             std::string jname = "free_" + child->getName();
             SimTK::Vec3 zeroVec(0.0);
-            Joint* free = new FreeJoint(jname, ground->getName(), child->getName());
+            Joint* free = new FreeJoint(jname, ground->getFullPathName(), child->getFullPathName());
+            free->upd_reverse() = mob.isReversedFromJoint();
             addJoint(free);
         }
         else{
-            Component* compToMoveOut = _components.at(m+nb);
+            Component* compToMoveOut = _propertySubcomponents.at(m+nb).get();
             // reorder the joint components in the order of the multibody tree
             Joint* jointToSwap = static_cast<Joint*>(mob.getJointRef());
-            it = std::find(_components.begin(), _components.end(), jointToSwap);
-            if (it != _components.end()){
+            auto it = std::find(_propertySubcomponents.begin(),
+                                _propertySubcomponents.end(), 
+                                SimTK::ReferencePtr<Component>(jointToSwap));
+            if (it != _propertySubcomponents.end()){
```

Comment:
There's nothing preventing someone for adding joints to their model as member or adopted subcomponents, right? So those should be handled as well. I think this can be taken care of in a separate PR, perhaps one that also addresses the issues that @klshrinidhi and I ran into when making the hopper device example (replacing access to joints with getComponentList).